library/core/src/macros/internals.rs

```diff
@@ -0,0 +1,85 @@
+use crate::{fmt, panic};
+
+#[cold]
+#[doc(hidden)]
+#[unstable(feature = "macros_internals", reason = "macros implementation detail", issue = "none")]
+#[track_caller]
```

Comment:
Do we need `#[track_caller]` here?

Reply:
I wasn't sure about this one, I put it because the `panic!` started in the inner function, and we want it to be reported near the `assert_eq!` call.